### PR TITLE
DEX-988 Add dummy frontend routes for url generation

### PR DESCRIPTION
**Pull request by karenc**
When we need to generate a frontend url, for example, to the password
reset form, we used to do:

```
urljoin(url_for('api.root', _external=True), f'/auth/code/{code}')
```

but now we can do:

```
url_for('frontend.auth-code', code=code, _external=True)
```

### CODE REVIEW

File: app/extensions/intelligent_agent/models.py [hunk 1]
```diff
@@ -536,7 +535,9 @@ def detection_complete(self):
             if self.owner and not self.owner.is_public_user():
                 ags = self.get_asset_group_sighting()
                 assert ags, f'no AGS for {self}'
-                url = full_api_url(f'pending-sightings/{str(ags.guid)}')
+                url = url_for(
+                    'frontend.pending-sightings', guid=str(ags.guid), _external=True
+                )
```
Comment (naknomum) [inline app/extensions/intelligent_agent/models.py:540]: yay thanks for going in and taking care of my use-cases!

Comment (karenc) [reply app/extensions/intelligent_agent/models.py:540]: Oh yeah, could you check that it's correct?  There's no tests so I'm not sure

File: app/utils.py [hunk 2]
```diff
@@ -168,17 +168,6 @@ def nlp_parse_complex_date_time(
...
-#  e.g. sighting.url() => url for sighting.  but i cant figure out how to get the api/route to pass:
-#       url_for('api.encounters_encounters') => 'http://localhost/api/v1/encounters/'
-#       url_for('api.sightings_sightings') => 'http://localhost/api/v1/sightings/'
-def full_api_url(suffix_path=None):
-    from urllib.parse import urljoin
-    from flask import url_for
-
-    return urljoin(url_for('api.root', _external=True), suffix_path)
-
-
```
Comment (naknomum) [inline app/utils.py:181]: thanks also for getting rid of the deprecated thing too.  :+1:

**Review by naknomum — APPROVED** [review]
lgtm

**Review by bluemellophone — COMMENTED** [review]
LGTM, the blueprints look great.  I'm glad we were able to register all of these endpoint routes on the existing frontend blueprint.

File: app/modules/app_ui.py [hunk 3]
```diff
@@ -44,10 +43,21 @@ def init_app(app):
...
 
+@frontend_blueprint.route('/', endpoint='root')
+@frontend_blueprint.route('/auth/code/<string:code>', endpoint='auth-code')
+@frontend_blueprint.route(
+    '/pending-sightings/<string:guid>', endpoint='pending-sightings'
+)
+@frontend_blueprint.route('/sightings/<string:guid>', endpoint='sightings')
+def dummy_view(*args, **kwargs):
+    """This view is created just for frontend route generation"""
+    return ''
```
Comment (bluemellophone) [inline app/modules/app_ui.py:54]: I would raise an explicit `RuntimeError` or `NotImplementedError` here, we want it to be clear that this code should never run

Comment (karenc) [reply app/modules/app_ui.py:54]: Ah ok, I can do that

**Pull request merged**